OpenRA.Mods.Common/Graphics/DefaultSpriteSequence.cs

```diff
@@ -144,6 +162,18 @@ public DefaultSpriteSequence(ModData modData, TileSet tileSet, SpriteCache cache
 							subLength = LoadField<int>(sd, "Length", 1);
 
 						combined = combined.Concat(subSprites.Skip(subStart).Take(subLength));
+
+						var subFrames = LoadField<int[]>(sd, "subFrames", null);
+						if (subFrames != null)
+						{
+							var fixedFrames = subFrames.Select(i => i + totalCount - subStart).ToArray();
+							if (Frames != null)
```

Comment:
This is mixing the overall (after combining) Frames with actions for the sub-frames sequences.  This doesn't look right to me.  The aggregate sequences should be building up a source collection of frames, just like if they were loaded from a single file, and then Frames indexes that later.